packages/playwright-test/src/runner.ts

```diff
@@ -675,29 +681,33 @@ function createTestGroups(rootSuite: Suite, workers: number): TestGroup[] {
       if (!withRequireFile) {
         withRequireFile = {
           general: createGroup(test),
-          parallel: [],
+          parallel: new Map(),
           parallelWithHooks: createGroup(test),
         };
         withWorkerHash.set(test._requireFile, withRequireFile);
       }
 
       let insideParallel = false;
-      let insideSerial = false;
+      let outermostMostSerialSuite: Suite | undefined;
       let hasAllHooks = false;
       for (let parent: Suite | undefined = test.parent; parent; parent = parent.parent) {
-        insideSerial = insideSerial || parent._parallelMode === 'serial';
-        // Serial cancels out any enclosing parallel.
-        insideParallel = insideParallel || (!insideSerial && parent._parallelMode === 'parallel');
+        if (parent._parallelMode === 'serial')
+          outermostMostSerialSuite = parent;
+        insideParallel = insideParallel || parent._parallelMode === 'parallel';
```

Comment:
let's comment that `describe.parallel` is never inside `describe.serial` - this is forced somewhere else